lib/opensusebasetest.pm

```diff
@@ -373,7 +372,7 @@ sub wait_boot {
         # booted so we have to handle that
         # because of broken firmware, bootindex doesn't work on aarch64 bsc#1022064
         push @tags, 'inst-bootmenu' if ((get_var('USBBOOT') and get_var('UEFI')) || (check_var('ARCH', 'aarch64') and get_var('UEFI')) || get_var('OFW'));
-        $self->handle_uefi_boot_disk_workaround if (get_var('MACHINE') =~ qr'aarch64' && get_var('BOOT_HDD_IMAGE') && !$in_grub);
+        $self->handle_uefi_boot_disk_workaround if (check_var('ARCH', 'aarch64') && get_var('UEFI') && get_var('BOOT_HDD_IMAGE') && !$in_grub);
```

Comment:
@richiejp I think you introduced the `get_var('MACHINE') =~ qr'aarch64'`-part

Reply:
I changed it to a regex of MACHINE from a `eq` of MACHINE, but in the commit I say that I am switching it to the ARCH variable. I have no idea how that happened, but obviously it should be using the ARCH var.

Reply:
You did it because of https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/2877#issuecomment-301412694 and I think that is still valid.

@ldevulder I would appreciate if you could change this back. Also, we still have the check for MACHINE in this file on another location. Now it is not consistent anymore.